### PR TITLE
feat(FEC-10783): create BP plugin

### PR DESCRIPTION
**Pull request by Yuvalke**
enable to use of broadpeak smartlib as a plugin which replacing our source by source we're getting from broadpeak smartlib.

### CODE REVIEW

File: src/bp-engine-decorator.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import {BroadPeak} from './bp-smartlib';
+
+/**
+ * Engine decorator for BroadPeak plugin.
+ * @class BpEngineDecorator
+ * @param {IEngine} engine - The HTML5 engine.
+ * @param {BroadPeak} plugin - The broadpeak plugin.
+ * @implements {IEngineDecorator}
+ */
+class BpEngineDecorator implements IEngineDecorator {
```
[the remark this answers is not in the document]
Comment (Yuvalke) [reply src/bp-engine-decorator.js]: Kept it with BP instead of BroadPeak

**Review by dan-ziv — CHANGES_REQUESTED** [review]
@Yuvalke LGTM but need to update README to integration with kaltura player and not playkit (see here https://github.com/kaltura/playkit-js-airplay)

File: src/bp-engine-decorator.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dan-ziv) [reply src/bp-engine-decorator.js]: It's your call, but naming usually should be consistent and not change from file to file.

[the remark this answers is not in the document]
Comment (Yuvalke) [reply src/bp-engine-decorator.js]: It's, also middleware use the shortcut and once it was broadpeak smartlib it also used BP instead broadpeak

File: README.md [hunk 2]
```diff
@@ -15,9 +15,7 @@ PlayKit JS Broadpeak Smartlib is written in [ECMAScript6], statically analysed u
 
 ### Prerequisites
 
-The plugin requires [PlayKit JS Player] to be loaded first.
-
-[playkit js player]: https://github.com/kaltura/playkit-js
+The plugin requires [Kaltura Player JS] to be loaded first.
```
Comment (dan-ziv) [inline README.md:18]: Title link is broken. need to add
[kaltura player js]: https://github.com/kaltura/kaltura-player-js

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**